src/hooks/keyboard-navigation.js

```diff
@@ -47,33 +69,40 @@ function isElementVisible(element) {
  * [2] https://www.w3.org/TR/wai-aria-practices/#keyboard
  *
  * @param {import('preact').RefObject<HTMLElement>} containerRef
- * @param {object} options
- *   @param {boolean} [options.autofocus] - Whether to focus the first element
- *     in the set of matching elements when the component is mounted
- *   @param {boolean} [options.horizontal] - Enable navigating elements using left/right arrow keys
- *   @param {boolean} [options.vertical] - Enable navigating elements using up/down arrow keys
- *   @param {string} [options.selector] - CSS selector which specifies the
- *     elements that navigation moves between
+ * @param {KeyboardNavigationOptions} options
  */
-export function useArrowKeyNavigation(
+export function useKeyboardNavigation(
   containerRef,
   {
+    active = true,
     autofocus = false,
-    horizontal = true,
-    vertical = true,
-    selector = 'a,button',
+    autofocusRef,
+    horizontal = false,
+    tab = false,
+    vertical = false,
+    selector = 'button, [href], input, select, textarea, [tabindex]',
   } = {}
 ) {
   useEffect(() => {
-    if (!containerRef.current) {
+    if (!active) {
+      return () => {};
+    }
+
+    if (active && !containerRef.current) {
       throw new Error('Container ref not set');
     }
-    const container = containerRef.current;
+
+    const container = /** @type {HTMLElement} */ (containerRef.current);
 
     const getNavigableElements = () => {
       const elements = /** @type {HTMLElement[]} */ (
         Array.from(container.querySelectorAll(selector))
       );
+      // If the container itself has a `tabIndex`, include it in the
+      // set of navigable elements
+      if (container.getAttribute('tabIndex')) {
+        elements.unshift(container);
+      }
```

Comment:
This allows the container itself to be part of the navigation sequence, allowing for cases in which we need to focus the dialog container itself on open (when no `focusRef` is provided).